converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/ChangeFeedProcessor.cs

```diff
@@ -64,72 +58,77 @@ public async Task ProcessAsync(TimeSpan pollIntervalDuringCatchup, CancellationT
-                    }
-                }
+                // Otherwise, process any new entries and increment the sequence
+                long maxSequence = changeFeedEntries.Count > 0 ? changeFeedEntries[^1].Sequence : state.SyncedSequence + DefaultLimit;
+                await ProcessChangeFeedEntriesAsync(changeFeedEntries, cancellationToken);
 
                 var newSyncState = new SyncState(maxSequence, Clock.UtcNow);
-
                 await _syncStateService.UpdateSyncStateAsync(newSyncState, cancellationToken);
+                state = newSyncState;
 
                 _logger.LogInformation("Processed DICOM events sequenced {SequenceId}-{MaxSequence}.", state.SyncedSequence + 1, maxSequence);
+                await Task.Delay(pollIntervalDuringCatchup, cancellationToken);
+            }
+        }
+    }
 
-                state = newSyncState;
+    private async Task ProcessChangeFeedEntriesAsync(IEnumerable<ChangeFeedEntry> changeFeedEntries, CancellationToken cancellationToken)
+    {
+        // Process each change feed as a FHIR transaction.
+        foreach (ChangeFeedEntry changeFeedEntry in changeFeedEntries)
+        {
+            try
+            {
+                if (!(changeFeedEntry.Action == ChangeFeedAction.Create && changeFeedEntry.State == ChangeFeedState.Deleted))
+                {
+                    await _fhirTransactionPipeline.ProcessAsync(changeFeedEntry, cancellationToken);
+                    _logger.LogInformation("Successfully processed DICOM event with SequenceID: {SequenceId}", changeFeedEntry.Sequence);
+                }
+                else
+                {
+                    _logger.LogInformation("Skip DICOM event with SequenceId {SequenceId} due to deletion before processing creation.", changeFeedEntry.Sequence);
+                }
+            }
+            catch (Exception ex) when (ex is FhirNonRetryableException or DicomTagException or TimeoutRejectedException)
+            {
+                string studyInstanceUid = changeFeedEntry.StudyInstanceUid;
+                string seriesInstanceUid = changeFeedEntry.SeriesInstanceUid;
+                string sopInstanceUid = changeFeedEntry.SopInstanceUid;
+                long changeFeedSequence = changeFeedEntry.Sequence;
 
-                await Task.Delay(pollIntervalDuringCatchup, cancellationToken);
+                ErrorType errorType = ErrorType.FhirError;
+
+                if (ex is DicomTagException)
+                {
+                    errorType = ErrorType.DicomError;
+                }
+                else if (ex is TimeoutRejectedException)
+                {
+                    errorType = ErrorType.TransientFailure;
+                }
+
+                await _exceptionStore.WriteExceptionAsync(changeFeedEntry, ex, errorType, cancellationToken);
+
+                _logger.LogError(
```

Comment:
These should in theory be customer errors that they can investigate, if I am not mistaken. So we probably wouldn't want to alert our DRI, but it could be useful to track. Still, I am unsure there is motivation at the moment for them